pkg/resource/model_package_group/hooks.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package model_package_group
+
+import (
+	"errors"
+	"time"
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	"github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	ErrModelPackageGroupDeleting     = errors.New("ModelPackageGroup in 'DELETING' state, cannot be modified or deleted")
+	ErrModelPackageGroupInProgress   = errors.New("ModelPackageGroup in 'INPROGRESS' state, cannot be modified or deleted")
+	ErrModelPackageGroupPending      = errors.New("ModelPackageGroup in 'PENDING' state, cannot be modified or deleted")
+	ErrModelPackageGroupDeleteFailed = errors.New("ModelPackageGroup in 'DELETEFAILED' state")
+)
+
+var (
+	// TerminalStatuses are the status strings that are terminal states for a
+	// ModelPackageGroup
+	TerminalStatuses = []v1alpha1.ModelPackageGroupStatus_SDK{
+		v1alpha1.ModelPackageGroupStatus_SDK_Deleting,
+	}
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		ErrModelPackageGroupDeleting,
+		5*time.Second,
+	)
+	requeueWaitWhileInProgress = ackrequeue.NeededAfter(
+		ErrModelPackageGroupInProgress,
+		5*time.Second,
+	)
+	requeueWaitWhilePending = ackrequeue.NeededAfter(
+		ErrModelPackageGroupPending,
+		5*time.Second,
+	)
+	requeueWaitWhileDeleteFailed = ackrequeue.NeededAfter(
+		ErrModelPackageGroupDeleteFailed,
+		5*time.Second,
+	)
+)
+
+// modelPackageGroupHasTerminalStatus returns whether the supplied SageMaker ModelPackageGroup is in a
+// terminal state
+func modelPackageGroupHasTerminalStatus(r *resource) bool {
```

Comment:
unused code, were you planning handle custom conditions in this PR?

Reply:
removed